src/block/Sugarcane.php

```diff
@@ -123,4 +128,22 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 
 		return false;
 	}
+
+	private function isSupportedByWater() : bool{
+		$down = $this->getSide(Facing::DOWN);
```

Comment:
`getSide(down)` is getting called outside of where this is used in both usages. Perhaps it might be worth passing it as a parameter instead of calling it a second time.